app/controllers/cases/offender_sar_controller.rb

```diff
@@ -391,7 +391,9 @@ def has_optional_flags?
     end
 
     def back_link_url
-      if @case.get_previous_step
+      if @case.get_previous_step && @case.get_previous_step == "subject-details"
```

Comment:
I think this logic should be done in `build_url_params_from_flags`, as that is responsible for adding query params